File: app/src/main/java/io/coderf/arklab/demo/viewmodel/DemoPagingViewModel.java
package io.coderf.arklab.demo.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;

import io.coderf.arklab.demo.api.ApiServiceHelper;
import io.coderf.arklab.demo.bean.NotificationMessageBean;
import io.coderf.arklab.demo.repository.KtDemoPagingRepositoryImpl;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import io.coderf.arklab.common.api.RepositoryFactory;
import io.coderf.arklab.common.base.BaseView;
import io.coderf.arklab.common.viewmodel.PagingViewModel;

/**
 * Created by fz on 2023/4/27 14:58
 * describe :
 */
@HiltViewModel
public class DemoPagingViewModel extends PagingViewModel<KtDemoPagingRepositoryImpl, NotificationMessageBean, BaseView> {
    @Inject
    ApiServiceHelper apiServiceHelper;

    @Inject
    public DemoPagingViewModel(@NonNull Application application) {
        super(application);
    }

    @Override
    protected KtDemoPagingRepositoryImpl createRepository() {
        return RepositoryFactory.create(KtDemoPagingRepositoryImpl.class,baseView,apiServiceHelper);
    }

}

File: app/src/main/java/io/coderf/arklab/demo/viewmodel/DemoPagingViewModel.kt
package io.coderf.arklab.demo.viewmodel

import android.app.Application
import dagger.hilt.android.lifecycle.HiltViewModel
import io.coderf.arklab.core.network.NetworkFlowPagingViewModel
import io.coderf.arklab.demo.api.ApiServiceHelper
import io.coderf.arklab.demo.bean.NotificationMessageBean
import io.coderf.arklab.demo.bean.RequestNotificationBean
import io.coderf.arklab.demo.repository.KtDemoPagingRepositoryImpl
import javax.inject.Inject

/**
 * 新闻分页 ViewModel（新版 [NetworkFlowPagingViewModel]，替代旧 Rx PagingViewModel）。
 *
 * @author fz
 * @version 1.0
 * @since 1.0
 * @updated 2026/9/12
 */
@HiltViewModel
class DemoPagingViewModel @Inject constructor(
    application: Application,
    private val apiServiceHelper: ApiServiceHelper
) : NetworkFlowPagingViewModel<KtDemoPagingRepositoryImpl, NotificationMessageBean, RequestNotificationBean>(
    application
) {

    override fun createRepository(): KtDemoPagingRepositoryImpl {
        return KtDemoPagingRepositoryImpl(apiServiceHelper)
    }

    override fun createPagingQuery() = RequestNotificationBean().apply {
        type = "5"
    }
}
